From 3f72a46490e4c464576fc9c21ead0039fc09bc52 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Tue, 8 May 2018 00:32:34 -0400
Subject: [PATCH] Revert "spi-nor: fslquad: add quad mode read for s25fs512s"

This reverts commit c6e770078025a650e4ced4e76cb002bd5e716214.

Since this patch did not exist in latest NXP SDK and would cause below error:
And actually for the "s25fs512s" spi nor id has already existed.

>mount -t ubifs ubi0:root /mnt
UBIFS (ubi0:0): default file-system created
UBIFS error (ubi0:0 pid 688): ubifs_read_node: bad node type (0 but expected 6)
UBIFS error (ubi0:0 pid 688): ubifs_read_node: bad node at LEB 0:0, LEB mapping status 1
Not a node, first 24 bytes:
00000000: 10 06 85 6b b8 83 01 00 00 00 00 00 00 00 00 10 00 00 06 00 00 00 00 00 ...k....................
CPU: 0 PID: 688 Comm: mount Not tainted 4.12.21-yocto-standard #1
Hardware name: Freescale LS1021A
[<8021000c>] (unwind_backtrace) from [<8020b86c>] (show_stack+0x20/0x24)
[<8020b86c>] (show_stack) from [<8063737c>] (dump_stack+0x78/0x94)
[<8063737c>] (dump_stack) from [<804c90c4>] (ubifs_read_node+0x1cc/0x264)
[<804c90c4>] (ubifs_read_node) from [<804c6508>] (ubifs_read_sb_node+0x64/0x88)
[<804c6508>] (ubifs_read_sb_node) from [<804c6c28>] (ubifs_read_superblock+0x6ac/0xc38)
[<804c6c28>] (ubifs_read_superblock) from [<804c5844>] (ubifs_mount+0x904/0x13b0)
[<804c5844>] (ubifs_mount) from [<8036b254>] (mount_fs+0x78/0x154)
[<8036b254>] (mount_fs) from [<803869d8>] (vfs_kern_mount+0x5c/0xf8)
[<803869d8>] (vfs_kern_mount) from [<803894c4>] (do_mount+0x644/0xab0)
[<803894c4>] (do_mount) from [<80389b80>] (SyS_mount+0x84/0xac)
[<80389b80>] (SyS_mount) from [<802079c0>] (ret_fast_syscall+0x0/0x40)

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/mtd/spi-nor/fsl-quadspi.c | 25 ++++++-------------------
 drivers/mtd/spi-nor/spi-nor.c     |  1 -
 2 files changed, 6 insertions(+), 20 deletions(-)

diff --git a/drivers/mtd/spi-nor/fsl-quadspi.c b/drivers/mtd/spi-nor/fsl-quadspi.c
index 33ecc27..50980a4 100644
--- a/drivers/mtd/spi-nor/fsl-quadspi.c
+++ b/drivers/mtd/spi-nor/fsl-quadspi.c
@@ -460,25 +460,12 @@ static void fsl_qspi_init_lut(struct fsl_qspi *q)
 			    LUT1(FSL_READ, PAD1, rxfifo),
 				base + QUADSPI_LUT(lut_base + 1));
 	} else if (nor->flash_read == SPI_NOR_QUAD) {
-		if (q->nor_size == 0x4000000) {
-			read_op = 0xEC;
-		qspi_writel(q,
-			LUT0(CMD, PAD1, read_op) | LUT1(ADDR, PAD4, addrlen),
-			base + QUADSPI_LUT(lut_base));
-		qspi_writel(q,
-			LUT0(MODE, PAD4, 0xff) | LUT1(DUMMY, PAD4, read_dm),
-			base + QUADSPI_LUT(lut_base + 1));
-		qspi_writel(q,
-			LUT0(FSL_READ, PAD4, rxfifo),
-			base + QUADSPI_LUT(lut_base + 2));
-		} else {
-			qspi_writel(q, LUT0(CMD, PAD1, read_op) |
-				    LUT1(ADDR, PAD1, addrlen),
-				    base + QUADSPI_LUT(lut_base));
-			qspi_writel(q, LUT0(DUMMY, PAD1, read_dm) |
-				    LUT1(FSL_READ, PAD4, rxfifo),
-				    base + QUADSPI_LUT(lut_base + 1));
-		}
+		qspi_writel(q, LUT0(CMD, PAD1, read_op) |
+			    LUT1(ADDR, PAD1, addrlen),
+				base + QUADSPI_LUT(lut_base));
+		qspi_writel(q, LUT0(DUMMY, PAD1, read_dm) |
+			    LUT1(FSL_READ, PAD4, rxfifo),
+				base + QUADSPI_LUT(lut_base + 1));
 	} else if (nor->flash_read == SPI_NOR_DDR_QUAD) {
 		/* read mode : 1-4-4, such as Spansion s25fl128s. */
 		qspi_writel(q, LUT0(CMD, PAD1, read_op)
diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index e19eef1..dbf3178 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -1075,7 +1075,6 @@ static int spi_nor_is_locked(struct mtd_info *mtd, loff_t ofs, uint64_t len)
 	{ "s25fl128s",	INFO6(0x012018, 0x4d0180, 64 * 1024, 256, SPI_NOR_QUAD_READ
 			| SPI_NOR_DDR_QUAD_READ) },
 	{ "s25fl129p0", INFO(0x012018, 0x4d00, 256 * 1024,  64, SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
-	{ "s25fs512s",  INFO6(0x010220, 0x4d0081, 256 * 1024, 256, SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ)},
 	{ "s25fl129p1", INFO(0x012018, 0x4d01,  64 * 1024, 256, SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
 	{ "s25sl004a",  INFO(0x010212,      0,  64 * 1024,   8, 0) },
 	{ "s25sl008a",  INFO(0x010213,      0,  64 * 1024,  16, 0) },
-- 
1.8.5.2.233.g932f7e4

